### DISCUSSION TITLE
Filter in the history popup?

### DISCUSSION THREAD

**Discussion by montoner0** [chrisant996/clink#588] [Q&A] [answered]
Hi. Is there a way to do filtering instead of incremental search in the `clink-popup-history`? It would be helpful when many similar command lines are in the history to narrow the list.

**Comment by chrisant996** [comment, marked as answer]
The usual way for doing that is to use the [fzf](https://github.com/junegunn/fzf) fuzzy finder tool.

There's a script for using fzf with Clink, and it's available in [clink-fzf](https://github.com/chrisant996/clink-fzf).  Or it's also included in [clink-gizmos](https://github.com/chrisant996/clink-gizmos) along with several other scripts.

**Comment by montoner0** [comment]
Thanks! Though it's not as nice as the clink native UI it's still better than nothing. 

BTW I've noticed a small error in the clink-fzf installation instructions. 
On step 2 it asks to "run `clink set fzf.exe_location <put_dir_name_here>` to tell Clink where to find the FZF program" that is not quite correct. 
Instead of `put_dir_name_here` there should be `put_exe_path_here`. Otherwise I was getting an error.

**Comment by chrisant996** [comment]
> Thanks! Though it's not as nice as the clink native UI it's still better than nothing.

It has far more filtering functionality than Clink would ever have.
The fzf UI is somewhat customizable as well (colors, preview window, height, and a bunch of other things).

You can make suggestions in the fzf repo for further enhancements to the fzf UI.

> BTW I've noticed a small error in the clink-fzf installation instructions. On step 2 it asks to "run `clink set fzf.exe_location <put_dir_name_here>` to tell Clink where to find the FZF program" that is not quite correct. Instead of `put_dir_name_here` there should be `put_exe_path_here`. Otherwise I was getting an error.

Thanks, I've updated the readme in the clink-fzf repo.
